corehq/apps/reports/standard/sms.py

```diff
@@ -260,6 +280,9 @@ def rows(self):
             if not message_type_filter(message_types):
                 continue
 
+            if reporting_locations_id and message.location_id not in reporting_locations_id:
```

Comment:
We should make reporting_locations_id be a dictionary instead of a list so that it performs much better for large lists of locations